packages/create-amplify/src/package-manager-controller/package_manager_controller_factory.ts

```diff
@@ -0,0 +1,117 @@
+import { existsSync as _existsSync } from 'fs';
+import { execa as _execa } from 'execa';
+import * as path from 'path';
+import { logger } from '../logger.js';
+import { NpmPackageManagerController } from './npm_package_manager_controller.js';
+import { PnpmPackageManagerController } from './pnpm_package_manager_controller.js';
+import { YarnClassicPackageManagerController } from './yarn_classic_package_manager_controller.js';
+import { YarnModernPackageManagerController } from './yarn_modern_package_manager_controller.js';
+import {
+  NpmProjectInitializer,
+  PnpmProjectInitializer,
+  YarnClassicProjectInitializer,
+  YarnModernProjectInitializer,
+} from '../project-initializer/index.js';
+
+export type DependencyType = 'dev' | 'prod';
+
+/**
+ *
+ */
+export abstract class PackageManagerController {
+  abstract installDependencies: (
+    packageNames: string[],
+    type: DependencyType
+  ) => Promise<void>;
+  abstract getPackageManagerProps: () => {
```

Comment:
This method should be removed. Instead we should have methods that tell the package manager what to do

Reply:
Done